ui-core/js/services/engine.ts

```diff
@@ -1372,14 +1372,36 @@ export const parseResponse = (
         Utils.replaceBrowserUrl(response);
     }
 
-    if (Utils.isEqual(response.invokeType, 'wait', true) ||
-        Utils.isEqual(response.invokeType, 'status', true)) {
+    if ((Utils.isEqual(response.invokeType, 'wait', true) ||
+        Utils.isEqual(response.invokeType, 'status', true)) &&
+        Utils.isNullOrUndefined(response.waitExpiresAt)) {
 
         ping(flowKey);
     }
 
+    if (!Utils.isNullOrUndefined(response.waitExpiresAt)) {
+        wait(new Date(response.waitExpiresAt), flowKey);
+    }
 };
 
+/**
+ * Wait until the current time exceeds the passed in date, then call back into the engine.
+ */
+export const wait = (until: Date, flowKey: string) => {
+    setTimeout(() => {
+        const state = State.getState(flowKey);
+        const options = State.getOptions(flowKey);
+
+        join(Utils.extractTenantId(flowKey),
+             Utils.extractFlowId(flowKey),
+             Utils.extractFlowVersionId(flowKey),
+             Utils.extractElement(flowKey),
+             state.id,
+             State.getAuthenticationToken(flowKey),
+             options);
+    }, until as any - Date.now())
```

Comment:
Setting a timeout to a specific time can be problematic in a couple of situations mentioned in these SO posts:

https://stackoverflow.com/questions/6032429/chrome-timeouts-interval-suspended-in-background-tabs

https://stackoverflow.com/questions/6346849/what-happens-to-settimeout-when-the-computer-goes-to-sleep

Just to be safe, we should probably set an interval loop and keep checking the current time to see if it has passed "until".